ggshield/cmd/secret/scan/archive.py

```diff
@@ -38,12 +46,17 @@ def archive_cmd(ctx: click.Context, path: str) -> int:  # pragma: no cover
             ignore_git=True,
         )
 
-        with click.progressbar(
-            length=len(files.files), label="Scanning", file=sys.stderr
-        ) as progressbar:
-
-            def update_progress(chunk: List[File]) -> None:
-                progressbar.update(len(chunk))
+        with Progress(
+            TextColumn("[progress.description]{task.description}"),
+            BarColumn(),
+            TaskProgressColumn(),
+            TextColumn("{task.completed} files scanned out of {task.total}"),
+            TimeRemainingColumn(),
+            console=Console(file=sys.stderr),
```

Comment:
This code is duplicated with a few variants in multiple places. Can you add a function to create a progress bar to `text_utils.py`? I am trying to group all output-related code there.